Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -2555,6 +2555,23 @@ std::optional<LayoutUnit> RenderBlock::firstLineBaseline() const
     return std::optional<LayoutUnit>();
 }
 
+std::optional<LayoutUnit> RenderBlock::lastLineBaseline() const
```

Comment:
This is very similar to RenderBlock::firstLineBaseline(), could it be shared, maybe by passing a lambda?

Reply:
I'm not sure if a lambda would have been easy to do since that would probably need to be done at the base class and not all derived implementations may need it. What I ended up doing was implementing RenderBox::firstInFlowChildBox and RenderBox::lastInFlowChildBox to help clean this up. I think it the overall readability and reusability, but let me know what you think!